optuna/pruners/hyperband.py

```diff
@@ -138,7 +159,7 @@ def __init__(
             )
 
             pruner = SuccessiveHalvingPruner(
-                min_resource=min_resource,
+                min_resource=1,
```

Comment:
Hmm, this PR always sets the `min_resource` value to `1`, but this doesn't seem to be suitable for tasks that report intermediate values at only a part of all steps.
For example, 
- [NAS-Bench-101](https://github.com/google-research/nasbench)
  - This benchmark dataset only has recorded intermediates values at 4, 12, 36, and 108 epochs.
  - In this case, I want to pass `min_resource=4`  to the `SuccessiveHalvingPruner` (via the `HyperbandPruner` construction).
- [Optuna for FFmpeg](https://gist.github.com/sile/8aa1ff7808dd55298f51dd70c8b83092#file-optimize-ffmpeg-py-L58) (Japanese)
  - In the objective function, intermediate values are reported at 100s, 300s, 900s positions of the input video.
  - Therefore, I constructed the pruner like `SuccessiveHalvingPruner(min_resource=100, reduction_factor=3))
` ([code](https://gist.github.com/sile/8aa1ff7808dd55298f51dd70c8b83092#file-optimize-ffmpeg-py-L85)).

Reply:
Thank you for you informative comments. I will fix the source code so that `min_resource` is appropriately and `min_early_stopping_rate` is always 0.